### PR TITLE
Add 1D signal b-splines to cupyx.scipy.signal

### PR DESCRIPTION
**Pull request by andfoy**
See https://github.com/cupy/cupy/issues/7403

### CODE REVIEW

File: cupyx/scipy/signal/_bsplines.py [hunk 1]
```diff
@@ -37,3 +40,353 @@ def sepfir2d(input, hrow, hcol):
...
+def _cubic(x):
+    x = cupy.asarray(x, dtype=float)
+    b = BSpline.basis_element(
+        cupy.asarray([-2, -1, 0, 1, 2]), extrapolate=False)
+    out = b(x)
+    out[(x < -2) | (x > 2)] = 0
+    return out
+
+
+def _coeff_smooth(lam):
```
Comment (takagi) [inline cupyx/scipy/signal/_bsplines.py:63]: Can we use kernel fusion to combine these expressions into an ElementwiseKernel?
https://docs.cupy.dev/en/stable/user_guide/kernel.html#kernel-fusion

File: cupyx/scipy/signal/_bsplines.py [hunk 2]
```diff
@@ -37,3 +40,353 @@ def sepfir2d(input, hrow, hcol):
...
+    xi = 1 - 96 * lam + 24 * lam * cupy.sqrt(3 + 144 * lam)
+    omeg = cupy.arctan2(cupy.sqrt(144 * lam - 1), cupy.sqrt(xi))
+    rho = (24 * lam - 1 - cupy.sqrt(xi)) / (24 * lam)
+    rho = rho * cupy.sqrt(
+        (48 * lam + 24 * lam * cupy.sqrt(3 + 144 * lam)) / xi)
+    return rho, omeg
+
+
+def _hc(k, cs, rho, omega):
```
Comment (takagi) [inline cupyx/scipy/signal/_bsplines.py:72]: ditto(kernel fusion)

File: cupyx/scipy/signal/_bsplines.py [hunk 3]
```diff
@@ -37,3 +40,353 @@ def sepfir2d(input, hrow, hcol):
...
+    return (cs / cupy.sin(omega) * (rho ** k) * cupy.sin(omega * (k + 1)) *
+            cupy.greater(k, -1))
+
+
+def _hs(k, cs, rho, omega):
```
Comment (takagi) [inline cupyx/scipy/signal/_bsplines.py:77]: ditto(kernel fusion)

File: cupyx/scipy/signal/_bsplines.py (hunk 1 above)
Comment (takagi) [inline cupyx/scipy/signal/_bsplines.py:63]: As `lam` is a Python scalar, would it be enough to compute these values on CPU?

**Comment by takagi** [comment]
Seems that these functions take some Python scalars, and computing coefficients on CPU then eventually transferring to GPU might be natural. Let me revisit next week.

**Comment by andfoy** [comment]
The heavy lifting work is done by calling ``sosfilt``, where the IIR computation part takes place

File: cupyx/scipy/signal/_bsplines.py (hunk 1 above)
Comment (takagi) [inline cupyx/scipy/signal/_bsplines.py:63]: If we keep this computation on GPU, I'd like to make it fused.
```suggestion
@cupy.fuse()
def _coeff_smooth(lam):
```

File: cupyx/scipy/signal/_bsplines.py (hunk 2 above)
Comment (takagi) [inline cupyx/scipy/signal/_bsplines.py:72]: ditto(`cupy.fuse()`)

File: cupyx/scipy/signal/_bsplines.py (hunk 3 above)
Comment (takagi) [inline cupyx/scipy/signal/_bsplines.py:77]: ditto(`@cupy.fuse()`)

**Review by takagi — COMMENTED** [review]
LGTM except for the minor comments.

**Comment by takagi** [comment]
Let me retry `cupy.win.cuda121` test.

**Review by takagi — APPROVED** [review]
LGTM!

**Comment by takagi** [comment]
Thanks, @andfoy!

**Pull request merged**